prow/config/branch_protection.go

```diff
@@ -241,15 +249,28 @@ func (c *Config) GetBranchProtection(org, repo, branch string) (*Policy, error)
 		return nil, err
 	}
 
-	return c.GetPolicy(org, repo, branch, *b)
+	return c.GetPolicy(githubClient, gitClient, org, repo, branch, *b)
 }
 
 // GetPolicy returns the protection policy for the branch, after merging in presubmits.
-func (c *Config) GetPolicy(org, repo, branch string, b Branch) (*Policy, error) {
+func (c *Config) GetPolicy(gitHubClient branchProtectionGitHubClient, gitClient *git.Client, org, repo, branch string, b Branch) (*Policy, error) {
 	policy := b.Policy
 
+	// This looks bad tokenwise but isn't, because Trigger does the very same request
+	// whenever it triggers Jobs for a PR that has branch as base. Consequently either:
+	// * The request is already cached because Trigger did it before, we dont have to pay a token
+	// * We do the request and it will get cached, making Trigger not have to pay a token
+	sha, err := gitHubClient.GetRef(org, repo, "heads/"+branch)
```

Comment:
...However on the first run of the branchprotector after this change, it may do some more `GetRef` calls, depending on how many branches no one ever used as base for a PR exist

Reply:
Using `git ls-remote` is a great idea! I wouldn't want to use that by default because I am too afraid its result will differ from what we get from the github API but for this particular case here we can certainly use it.

Reply:
On second thought though, this will impact the runtime quite a bit. When playing around, a `git ls-remote` takes 1-1.5S, if we do this once for every branch that may take a moment. WDYT?